blocks/firefly-gallery/firefly-gallery.js

```diff
@@ -108,21 +113,51 @@ async function addCards(cardContainer, images) {
       cardDetails.append(cardFooter);
       card.append(cardDetails);
       cardContainer.append(card);
+
+      card.addEventListener('click', () => {
+        window.open(communityUrl, '_self');
+      });
+
+      favorite.addEventListener('click', async (e) => {
+        e.preventDefault();
+        if (window.adobeIMS.isSignedInUser()) {
+          const accessToken = window.adobeIMS.getAccessToken();
+          const imageId = image.urn.split(':').pop();
+          const url = FAVOURITE_URL.replace('$', imageId);
+          const headers = new Headers({
+            'X-Api-Key': 'alfred-community-hubs',
+            'community_id': 'ff_community',
+            'Authorization': `Bearer ${accessToken.token}`,
+            'content-type': 'application/json',
+          });
+          const resp = await fetch(url, {
+            method: 'PUT',
+            headers,
+            mode: 'cors',
+          });
+          if (resp.ok) {
+            favorite.classList.add('liked');
+          }
+        }
+      });
     }
   });
+  // set height of container
+  cardContainer.style.height = `${heightOfContainer}px`;
   GETTING_IMAGES = false;
 }
 
 export default async function decorate(block) {
-  let size = SHORT_GALLERY_SIZE;
```

Comment:
why pull in 48 when we need only 20 for the short gallery size

Reply:
@amol-anand On the live site, the overflow effect is achieved this way. There, it renders 48 images and shows only ~ 20 images, I am trying achieve similar effect here by setting the height of the container equal to height of first 5 images (as the container is a 4x5 table) and setting overflow: hidden for the container.